opl/investigator/check.py

```diff
@@ -42,6 +42,52 @@ def _check_by_min_max(data, value):
     return lower_boundary <= value <= upper_boundary, info
 
 
+def _check_by_max(data, value):
+    logging.debug(f"data={data} and value={value}")
+    mean = statistics.mean(data)
+    lower_boundary = float(mean - (mean - min(data)))
+    upper_boundary = float(mean + (max(data) - mean))
+    logging.info(
+        f"value={value}, data len={len(data)} mean={mean:.03f}, i.e. boundaries={lower_boundary:.03f}--{upper_boundary:.03f}"
+    )
+    info = collections.OrderedDict(
+        [
+            ("method", inspect.stack()[1][3]),
+            ("value", value),
+            ("data len", len(data)),
+            ("data mean", mean),
+            ("data min", float(min(data))),
+            ("data max", float(max(data))),
+            ("lower_boundary", lower_boundary),
+            ("upper_boundary", upper_boundary),
+        ]
+    )
+    return value <= upper_boundary, info
+
+
+def _check_by_min(data, value):
+    logging.debug(f"data={data} and value={value}")
+    mean = statistics.mean(data)
+    lower_boundary = float(mean - (mean - min(data)))
+    upper_boundary = float(mean + (max(data) - mean))
```

Comment:
same idea, maybe `upper_boundary = None` or set it to `-1` or `99999999999` something big to communicate no upper boundary depending on what framework accepts w/o freaking out